### ISSUE TITLE
HTTP 500 when using external authorization plugins and GoCD site urls are not set

### ISSUE THREAD

**Issue by pkovtunov** [gocd/gocd#10036]
##### Issue Type

- Bug Report

##### Summary

When using GoCD GitLab or GitHub authentication plugins, the server site URLs must be set properly in the server configuration, otherwise the GoCD will fail with HTTP 500.

##### Environment

Discovered on 21.2.0, Java 15.
I suppose it is not related to the GoCD or Java version.

###### Basic environment details

* Go Version: 21.2.0 
* JAVA Version: 15.0.2
* OS: Linux 4.19.0-17-amd64

##### Steps to Reproduce

1. Put a GitLab or GitHub plugin JAR into the appropriate folder and restart the server
2. Create a new authorization configuration with the desired plugin
3. Make sure that both server site urls are not set OR at least the second one (Secure site URL) is not set
4. Try to login via plugin

##### Expected Results

Should be redirected to the external IDP properly

##### Actual Results

GoCD throws an HTTP500

##### Possible Fix

Use at least the site url which is set or something default, like http://localhost:8153

##### Log snippets

In the go-server.log you can see following:

```
2022-01-03 10:25:48,470 WARN  [qtp679525351-29] HttpChannel:673 - handleException /go/plugin/cd.go.authorization.gitlab/login java.net.MalformedURLException: no protocol:
```

**Comment by chadlwilson** [comment]
I'm wondering whether this validation should be done by the plugins themselves. Do you have the full stack trace/error from one of the plugins?

**Comment by pkovtunov** [comment]
There is no log output from the plugin at all, because the 500 happens even before the plugin code is called. As I was fighting with this bus, I had a remote debugging on for my GoCD and some breakpoints set. So, it happens somewhere in GoCD code itself, before the main plugin class is called.

**Comment by chadlwilson** [comment]
Sorry, let me rephrase - do you have the full stack trace/error on the server from `go-server.log` that happens when you use one of the plugins?

The single error line above makes it difficult to narrow in on the problem. It's a lot easier if you include a fuller error from the server log so it's clear where the `MalformedURLException` is being experienced.

**Comment by pkovtunov** [comment]
np ;) This is the only line emmited in the log, when the error occures. I'm not sure for now, which configuration I should add to logback-include.xml to bump up the verbosity - would you please give me a hint?

**Comment by chadlwilson** [comment]
I'm not sure in this case. :-) I was hoping the stack trace might give me ideas for places to start looking as I haven't looked at the authz plugins before. No worries - at some point I can reproduce it from first principles and see the issue :-)

**Comment by tewfik-ghariani** [comment]
This has already been solved per https://github.com/gocd-contrib/gitlab-oauth-authorization-plugin/pull/119

The URLs had to be defined in the server configuration by the admin

**Comment by chadlwilson** [comment]
The documentation helps, but it'd be better if the system and/or the validation prevented you getting to a 500 error in the first place, especially if the problem is common to all such plugins since documenting it individually in each plugin is not so scalable.
